pandas/tests/resample/test_datetime_index.py

```diff
@@ -1463,3 +1464,37 @@ def f(data, add_arg):
         result = df.groupby("A").resample("D").agg(f, multiplier)
         expected = df.groupby("A").resample('D').mean().multiply(multiplier)
         assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize('first,last,offset,exp_first,exp_last', [
+        ('19910905', '19920406', 'D', '19910905', '19920407'),
+        ('19910905 00:00', '19920406 06:00', 'D', '19910905', '19920407'),
+        ('19910905 06:00', '19920406 06:00', 'H', '19910905 06:00',
+         '19920406 07:00'),
+        ('19910906', '19920406', 'M', '19910831', '19920430'),
+        ('19910831', '19920430', 'M', '19910831', '19920531'),
+        ('1991-08', '1992-04', 'M', '19910831', '19920531'),
+    ])
+    def test_get_timestamp_range_edges(self, first, last, offset,
+                                       exp_first, exp_last):
+        first = pd.Period(first)
+        first = first.to_timestamp(first.freq)
+        last = pd.Period(last)
+        last = last.to_timestamp(last.freq)
+
+        exp_first = pd.Timestamp(exp_first, freq=offset)
+        exp_last = pd.Timestamp(exp_last, freq=offset)
+
+        offset = pd.tseries.frequencies.to_offset(offset)
+        result = _get_timestamp_range_edges(first, last, offset)
+        expected = (exp_first, exp_last)
+        assert result == expected
+
+    @pytest.mark.parametrize('first,last', [
+        (pd.Period('1991-09-05'), pd.Timestamp('1992-04-06')),
+        (pd.Timestamp('1991-09-05'), pd.Period('1992-04-06')),
+        (pd.Period('1991-09-05'), pd.Period('1992-04-06')),
+        (pd.Timestamp('1991-09-05'), '1992-04-06'),
+    ])
+    def test_get_period_range_edges_bad_input(self, first, last):
+        with pytest.raises(TypeError, match='instances of type Timestamp'):
+            _get_timestamp_range_edges(first, last, pd.offsets.Day())
```

Comment:
Actually, since these functions are only used once and are private, can these hit be simply calling `Series.resample` or `DataFrame.resample`? If they cannot, we may not need to check that first and last are the same scalar type. (same as the other test)